auth/provider/github_profile.go

```diff
@@ -45,12 +46,19 @@ func (gh *GitHub) GetProfile(accessToken string) (*GitHubProfile, error) {
 
 		bytesOut, _ := ioutil.ReadAll(res.Body)
 
-		unmarshalErr := json.Unmarshal(bytesOut, profile)
+		unmarshalErr := json.Unmarshal(bytesOut, &githubProfile)
 		if unmarshalErr != nil {
 			return profile, unmarshalErr
 		}
 	}
 
+	profile.TwoFactor = githubProfile.TwoFactor
```

Comment:
What's this intermediate entity for?

Reply:
I didn't want to interfere a lot with your code (refactoring) because I know you are not happy about that, and instead having two different abstractions for oauthProvider I have added small layer which gives us more flexibility for potential future improvements